pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -227,13 +233,25 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	deviceMountable := dsw.isDeviceMountableVolume(volumeSpec)
 
 	if _, volumeExists := dsw.volumesToMount[volumeName]; !volumeExists {
+		var sizeLimit int64
+		sizeLimit = 0
+		isLocal, _ := limits.IsLocalEphemeralVolume(pod, volumeSpec.Name())
+		if isLocal {
+			_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+			ephemeralStorageLimit := podLimits[v1.ResourceEphemeralStorage]
```

Comment:
We should make sure the podLimit exists within the map by adding an `ok` validation.